nebula-logger/core/main/logger-engine/classes/LogEntryEventBuilder.cls

```diff
@@ -603,6 +603,50 @@ global with sharing class LogEntryEventBuilder {
         return this;
     }
 
+    /**
+     * @description Sets a field values on the builder's `LogEntryEvent__e` record
+     * @param  field      The `Schema.SObjectField` token of the field to populate
+     *                    on the builder's `LogEntryEvent__e` record
+     * @param  fieldValue The `Object` value to populate in the provided field
+     * @return The same instance of `LogEntryEventBuilder`, useful for chaining methods
+     */
+    global LogEntryEventBuilder setField(Schema.SObjectField field, Object fieldValue) {
+        if (this.shouldSave() == false) {
+            return this;
+        }
+
+        return this.setFields(new Map<Schema.SObjectField, Object>{ field => fieldValue });
+    }
+
+    /**
+     * @description Sets multiple field values on the builder's `LogEntryEvent__e` record
+     * @param  fieldToValue An instance of `Map<Schema.SObjectField, Object>` containing the
+     *                      the fields & values to populate the builder's `LogEntryEvent__e` record
+     * @return The same instance of `LogEntryEventBuilder`, useful for chaining methods
+     */
+    @SuppressWarnings('PMD.AvoidDebugStatements')
+    global LogEntryEventBuilder setFields(Map<Schema.SObjectField, Object> fieldToValue) {
```

Comment:
TODO decide if the method name should just be `set()` (or `put()`? 🤔 ) - the name would simpler & consistent for the overload/avoids the different names `setField()` and `setFields()`